code/modules/bitrunning/components/glitch.dm

```diff
@@ -0,0 +1,89 @@
+/datum/component/glitch
+	/// Ref of the spawning forge
+	var/datum/weakref/forge_ref
+
+/datum/component/glitch/Initialize(obj/machinery/quantum_server/server, obj/machinery/byteforge/forge)
+	if(!isliving(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	RegisterSignal(forge, COMSIG_MACHINERY_POWER_RESTORED, PROC_REF(on_forge_power_restored))
+	RegisterSignals(forge, list(COMSIG_MACHINERY_BROKEN, COMSIG_MACHINERY_POWER_LOST), PROC_REF(on_forge_broken))
+	forge_ref = WEAKREF(forge)
+
+	var/mob/living/owner = parent
+	server.spawned_threat_refs.Remove(WEAKREF(owner)) // so the server doesn't dust us
+
+	owner.faction.Cut()
+	owner.faction += list(ROLE_GLITCH)
+
+	var/current_max = owner.maxHealth + ROUND_UP(server.threat * 0.2)
+	owner.maxHealth = clamp(current_max, 200, 500)
+	owner.fully_heal()
+
+	owner.AddElement(/datum/element/digital_aura)
+
+/datum/component/glitch/RegisterWithParent()
+	RegisterSignals(parent, list(COMSIG_LIVING_STATUS_UNCONSCIOUS, COMSIG_LIVING_DEATH), PROC_REF(on_death))
+
+/datum/component/glitch/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_LIVING_DEATH, PROC_REF(on_death))
+
+/// Sakujo
+/datum/component/glitch/proc/dust_mob()
+	if(QDELETED(parent))
+		return
+
+	var/mob/living/owner = parent
+	owner.dust()
+	qdel(src)
```

Comment:
is this required won't dusting the mob do this on its own? or are you just being safe (do we want to qdel this if the dust fails?)

Reply:
It does I'm just being super paranoid

Reply:
would we want it qdel'd if the dust didn't work out?